doc/developer-notes.md

```diff
@@ -189,9 +191,18 @@ Not OK (used plenty in the current source, but not picked up):
 A full list of comment syntaxes picked up by Doxygen can be found at http://www.doxygen.nl/manual/docblocks.html,
 but the above styles are favored.
 
+Other recommendations:
+
+- Avoid redundant type and in/out information in function descriptions
```

Comment:
Can you give an example for all of these? (like the cases above)

Reply:
Added some examples and simplified recommendations.